File: src/main/java/com/lalkalol/mujahid/audio/GuildMusicManager.java
package com.lalkalol.mujahid.audio;

import dev.arbjerg.lavalink.client.Link;
import dev.arbjerg.lavalink.client.player.LavalinkPlayer;

public class GuildMusicManager {
    private final long guildId;
    private final LavalinkManager manager;
    private final TrackScheduler scheduler;

    public GuildMusicManager(long guildId, LavalinkManager manager) {
        this.guildId = guildId;
        this.manager = manager;
        this.scheduler = new TrackScheduler(this);
    }

    public long getGuildId() {
        return guildId;
    }

    public LavalinkManager getManager() {
        return manager;
    }

    public TrackScheduler getScheduler() {
        return scheduler;
    }

    public Link getLink() {
        return manager.getClient().getOrCreateLink(guildId);
    }

    public Link getCachedLink() {
        return manager.getClient().getLinkIfCached(guildId);
    }

    public LavalinkPlayer getPlayer() {
        Link link = getCachedLink();
        return link != null ? link.getCachedPlayer() : null;
    }
}

File: src/main/kotlin/com/lalkalol/mujahid/audio/GuildMusicManager.kt
package com.lalkalol.mujahid.audio

import dev.arbjerg.lavalink.client.Link
import dev.arbjerg.lavalink.client.player.LavalinkPlayer

class GuildMusicManager(
    val guildId: Long,
    private val manager: LavalinkManager,
) {
    val scheduler: TrackScheduler = TrackScheduler(this)

    fun getManager(): LavalinkManager = manager

    fun getLink(): Link = manager.client.getOrCreateLink(guildId)

    fun getCachedLink(): Link? = manager.client.getLinkIfCached(guildId)

    fun getPlayer(): LavalinkPlayer? = getCachedLink()?.cachedPlayer
}
